client/landing/stepper/hooks/use-site-transfer/query.ts

```diff
@@ -75,8 +79,10 @@ export const useSiteTransferStatusQuery = ( siteId: number | undefined, options?
 			};
 		},
 		refetchOnWindowFocus: false,
-		refetchInterval: ( { state } ) =>
-			shouldRefetch( state.data?.status ) ? REFETCH_TIME : false,
+		refetchIntervalInBackground: !! options?.refetchIntervalInBackground,
+		refetchInterval:
+			options?.refetchInterval ||
+			( ( { state } ) => ( shouldRefetch( state.data?.status ) ? REFETCH_TIME : false ) ),
```

Comment:
Looking closer at the existing logic here, I wonder why we need to override `refetchInterval`. Doesn't the existing `shouldRefetch` condition already take care of things here..?

Reply:
There are some nuances, for example existsing refetchInterval won't be working in case if transfer to Atomic wasn't started firing the initial page rendering (`! status`) - https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/hooks/use-site-transfer/query.ts#L49

And I faced some extra cases, but don't remember exactly which cases. Our main goal - not render "Activate" button since clicking on it when activation process was already started - makes mess at the backend. So with static refetchInterval we are make sure that we always checking for updates, to avoid rendering "Activate" button when the process is started.

Reply:
> won't be working in case if transfer to Atomic wasn't started firing the initial page rendering

Could you clarify this part? Do you mean that the existing logic won't make the hook refetch if the transfer was started before we load the page? 

Keeping the `refetchIntervalInBackground` option makes sense, but it seems to me like the existing `refetchInterval` logic here _should_ work as it is. I know my suggestion is what prompted you to look into this, though 🙇

Reply:
> Could you clarify this part?

@fredrikekelund the case is described in `Testing Instructions -> Manually turning on Hosting Features`:
1) The user opens the "Hosting Features" page, sees the "Activate" button, doesn't click on it and leaves the tab open. NB At this moment `transferStatus === 'none'`, since transfer isn't started yet, hense we *don't* trigger refetching by timeout, due to [this condition](https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/hooks/use-site-transfer/query.ts#L55).
2) The user opens the second tab with the same page of "Hosting features" and in this tab they click on "Activate" button.

As a result, the first tab won't refetch data by interval and if the user comes back to that page and click on "Activate" button - they will get errors. It was the initial issue that Angela asked to handle in the "Hosting Refinements" project pcYYuD-1Pw-p2.

That's why I was introducing `refetchInterval: 3000` in this PR, to have refetching always, even when the status is "none".
But this PR is closed and I just created follow up PR just to cover this case and I decided that we can actually simplify the fix and make it in a more appropriate way - https://github.com/Automattic/wp-calypso/pull/95666/files#diff-0ca01c7291c804cf8b6e8e9ca672e1c9d635149e5af0bc1b439c9d4353445ed5R59-R63